src/targets/gpu/device/include/migraphx/gpu/device/launch.hpp

```diff
@@ -81,18 +81,25 @@ inline auto launch(hipStream_t stream, index_int global, index_int local)
         using f_type = decltype(f);
         dim3 nblocks(global / local);
         dim3 nthreads(local);
+        /*
+        hipGetLastError() returns error for the first failed HIP call that happened previously.
+        MIGraphX calls into various backend libraries and failed HIP calls can also happen there.
+        Calling hipGetLastError() would reset error code to hipSuccess, so that inside MIGraphX
+        failed call to hipLaunchKernelGGL() can be captured.
+        */
+        hipError_t flush_call = hipGetLastError();
+        (void)(flush_call);
         // cppcheck-suppress UseDeviceLaunch
         hipLaunchKernelGGL((launcher<f_type>), nblocks, nthreads, 0, stream, f);
         hipError_t kernel_launch_status = hipGetLastError();
         if(kernel_launch_status != hipSuccess)
         {
             std::string message = hipGetErrorString(kernel_launch_status);
-            if(not contains(get_targets(), get_device_name()))
+            if(not contains(get_targets(), get_gfx_name()))
```

Comment:
If the target ID is different then this message wouldn't show. For example, if this is for gfx908 with xnack on and it was built with xnack off(which is the default), then we wont see the error message.

Reply:
we are not specifying `XNACK` and `SRAMECC` flags for the GPU_TARGETS.

Reply:
> we are not specifying XNACK and SRAMECC flags for the GPU_TARGETS.

But users can(and is necessary) when they want to build for xnack off.

Reply:
Removed that change.